src/geowombat/core/api.py

```diff
@@ -516,42 +519,52 @@ def __init__(
                         dtype=dtype,
                         netcdf_vars=netcdf_vars,
                         nodata=nodata,
-                        **kwargs
+                        **kwargs,
                     )
 
                 else:
-                    if 'chunks' in kwargs and not isinstance(kwargs['chunks'], dict):
-                        logger.exception('  The chunks should be a dictionary.')
+                    if "chunks" in kwargs and not isinstance(kwargs["chunks"], dict):
+                        logger.exception("  The chunks should be a dictionary.")
                         raise TypeError
 
                     with xr.open_dataset(filename, **kwargs) as src:
-                        self.data = src.to_array(dim='band')
+                        self.data = src.to_array(dim="band")
+                    # Ensure the filename attribute gets updated as the NetCDF file
+                    self.data = self.data.assign_attrs(**{"filename": str(filename)})

```

Comment:
This is a fix to ensure the NetCDF filename replaces the filename attribute stored in the NetCDF file.